test/e2e/test/version.go

```diff
@@ -7,18 +7,39 @@ package test
 import (
 	"testing"
 
-	"github.com/stretchr/testify/require"
-	"k8s.io/apimachinery/pkg/util/version"
+	"github.com/elastic/cloud-on-k8s/pkg/controller/common/version"
 )
 
-// MinVersionOrSkip run this test only if the server has at least version v.
-func MinVersionOrSkip(t *testing.T, v string) {
-	info, err := ServerVersion()
-	require.NoError(t, err)
+// Elastic Stack versions used in the E2E tests
+const (
+	// Minimum version for 6.8.x tested with the operator
+	MinVersion68x = "6.8.5"
+	// Current latest version for 7.x
+	LatestVersion7x = "7.6.0" // version to synchronize with the latest release of the Elastic Stack
+)
 
-	min := version.MustParseGeneric(v)
-	actual := version.MustParseGeneric(info.GitVersion)
-	if !actual.AtLeast(min) {
+// SkipInvalidUpgrade skips a test that would do an invalid upgrade.
+func SkipInvalidUpgrade(t *testing.T, srcVersion string, dstVersion string) {
+	isValid, err := isValidUpgrade(srcVersion, dstVersion)
+	if err != nil {
+		t.Fatal("Fail to parse Elastic Stack version", "err", err.Error())
```

Comment:
Nit: I think this would just concatenate everything together and come out as `Fail to parse Elastic Stack versionerrerror message...`.  Maybe `isValidUpgrade` should return a wrapped error to provide more information. I don't know whether that's overkill for a test function though.

```suggestion
		t.Fatalf("Failed to determine the validity of the upgrade path: %v", err)
		...
func isValidUpgrade(from string, to string) (bool, error) {
	srcVer, err := version.Parse(from)
	if err != nil {
		return false, fmt.Errorf("failed to parse version '%s': %w", from, err)
	}
	...
}
```